cuegui/cuegui/JobMonitorTree.py

```diff
@@ -285,6 +285,12 @@ def addJob(self, job, timestamp=None, loading_from_config=False):
                         dep = self.__menuActions.jobs(
                         ).getRecursiveDependentJobs([newJobObj],
                                                     active_only=active_only)
+
+                        # Remove dependent if it has the same name as the job
```

Comment:
Could you explain this fix a bit more? In the PR you say "jobs without dependency are being omitted" but it's not clear to me how this change solves the problem. Just trying to understand.

Reply:
Hi @bcipriano 

The changes in this new PR is a fix for a bug created in the following PRs:

Dependent feature
https://github.com/AcademySoftwareFoundation/OpenCue/pull/815

Add kill dependent and group dependent features
https://github.com/AcademySoftwareFoundation/OpenCue/pull/1115

There was a feature implementation that added the group-dependent features on the CueGUI.

To reproduce the error go to:

CueGUI > Monitor Jobs and check the checkbox "[x] Group Dependent"

- Some jobs that had dependents were not shown in the Monitor Job.

However, if you uncheck the checkbox "[] Group Dependent" the jobs are displayed correctly.


After get the dependent jobs list that wrongly included the parent job name:

JobMonitorTree.py > class JobMonitorTree(cuegui.AbstractTreeWidget.AbstractTreeWidget): > def addJob(self, job, timestamp=None, loading_from_config=False):
```
dep = self.__menuActions.jobs()
.getRecursiveDependentJobs([newJobObj], active_only=active_only)
```


The bug was happening because the this for:

JobMonitorTree.py > class JobMonitorTree() > def addJob():
```
for j in self.__reverseDependents:
                        if j in self.__load:
                            del self.__load[j]
```

The ```__reverseDependents``` had the name of the parent job and when the ```del self.__load[j]``` is executed, the CueGui was removing the parent job and all the dependent jobs, resulting on missing jobs on MonitorJobs.

Reply:
Hi @bcipriano 

I sent a new version using list comprehension to avoid many nested blocks and follow the project code style.

https://github.com/AcademySoftwareFoundation/OpenCue/pull/1312/commits/57afbb3291ca7aa48f87cde63afb373ab276743c


Lint Python Code error:

https://github.com/AcademySoftwareFoundation/OpenCue/actions/runs/5942566910/job/16118316707

```
Running lint for cuegui/...
************* Module cuegui.JobMonitorTree
cuegui/JobMonitorTree.py:262:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
```